File: PyProject/geeksforgeeks/select_sort.py
# -*- coding:utf-8 -*-
# author: youdi

from typing import List


def select_sort(nums: List[int]) -> List[int]:
    nums_length = len(nums)
    for i in range(nums_length):
        min_index = i
        for j in range(i + 1, nums_length):
            if nums[j] < nums[min_index]:
                min_index = j
        nums[min_index], nums[i] = nums[i], nums[min_index]
    return nums


if __name__ == '__main__':
    A = [64, 25, 12, 22, 11]
    B = select_sort(A)
    print(B)

File: GoProject/leetcode/sort/history/select_sort.go
package main

func selectionSort(nums []int) {
	l := len(nums)
	var minIndex int
	for i := 0; i < l-1; i++ {
		minIndex = 1
		for j := i; j < l; j++ {
			if nums[minIndex] > nums[j] {
				minIndex = j
			}
		}
		nums[i], nums[minIndex] = nums[minIndex], nums[i]
	}

}

func main() {
	a := []int{5, 2, 3, 1}
	bubbleSort(a)
	for _, v := range a {
		println(v)
	}
}
